crates/subspace-farmer/src/commands/farm.rs

```diff
@@ -45,32 +39,37 @@ pub async fn farm(base_directory: PathBuf, ws_server: &str) -> Result<()> {
 
     let identity = Identity::open_or_create(&base_directory)?;
 
-    match future::select(
-        {
-            let client = client.clone();
-            let plot = plot.clone();
-            let commitments = commitments.clone();
-            let public_key = identity.public_key();
+    // start the farming task
+    // right now the instance is unused, however, if we want to call stop the process
+    // we can just drop the instance, and it will be stopped magically :)
+    let _farming_instance = Farming::start(
+        plot.clone(),
+        commitments.clone(),
+        client.clone(),
+        identity.clone(),
+    );
+
+    // start the background plotting
+    // NOTE: THIS WILL CHANGE IN THE UPCOMING PR
+    let plotting_result = {
+        let client = client.clone();
+        let plot = plot.clone();
+        let commitments = commitments.clone();
+        let public_key = identity.public_key();
 
-            Box::pin(async move {
-                background_plotting(client, plot, commitments, object_mappings, &public_key).await
-            })
-        },
-        Box::pin(
-            async move { subscribe_to_slot_info(&client, &plot, &commitments, &identity).await },
-        ),
-    )
-    .await
-    {
-        Either::Left((result, _)) => match result {
-            Ok(()) => {
-                info!("Background plotting finished successfully");
-
-                Ok(())
-            }
-            Err(error) => Err(anyhow!("Background plotting error: {}", error)),
-        },
-        Either::Right((result, _)) => result.map_err(Into::into),
+        Box::pin(async move {
+            background_plotting(client, plot, commitments, object_mappings, &public_key).await
+        })
+    }
+    .await;
```

Comment:
This can be simplified further to the following now:
```rust
let public_key = identity.public_key();
let plotting_result = background_plotting(client, plot, commitments, object_mappings, &public_key).await;
```

Reply:
Oh yes. However, farm.rs will change significantly in the next PR, so will do this in the next PR, thanks!